### PR TITLE
refactor(Core/Conditions): Add unused conditions as placeholders

### PR DESCRIPTION
**Pull request by Kitzunu**
## CHANGES PROPOSED:
* Turn CONDITION_TERRAIN_SWAP into a placeholder

* Turn CONDITION_DIFFICULTY_ID into a placeholder

* Turn CONDITION_SOURCE_TYPE_TERRAIN_SWAP into a placeholder

* Turn CONDITION_SOURCE_TYPE_PHASE into a placeholder

* Turn CONDITION_SOURCE_TYPE_GRAVEYARD into a placeholder

## HOW TO TEST THE CHANGES:

Not needed since they are not used, just make sure the build passes

## Target branch(es):
- [x] Master

 
## How to test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Conditions/ConditionMgr.cpp [hunk 1]
```diff
@@ -1259,32 +1259,33 @@ bool ConditionMgr::isSourceTypeValid(Condition* cond)
...
-        case CONDITION_SOURCE_TYPE_TERRAIN_SWAP:
-        case CONDITION_SOURCE_TYPE_PHASE:
-        case CONDITION_SOURCE_TYPE_GRAVEYARD:
+        case CONDITION_SOURCE_TYPE_UNUSED_20:
+        case CONDITION_SOURCE_TYPE_UNUSED_25:
+        case CONDITION_SOURCE_TYPE_UNUSED_26:
+        case CONDITION_SOURCE_TYPE_UNUSED_27:
         {
-            sLog->outErrorDb("ConditionSourceType %u in `condition` table is not supported on 3.3.5a, ignoring.", uint32(cond->SourceType));
+            sLog->outErrorDb("ConditionSourceType = (%u) in `conditions` is not in use. SourceEntry = (%u), skipped", uint32(cond->SourceType), cond->SourceEntry);
```
Comment (BarbzYHOOL) [inline src/server/game/Conditions/ConditionMgr.cpp:1273]: is not in use?? that doesn't seem to convey the right information to the user

user will think "ah i need to change this setting to make it 'in use'" or smthg like that

what do you mean by "is not in use" ?? i think you mean "is not implemented yet" which is much clearer

File: src/server/game/Conditions/ConditionMgr.cpp [hunk 2]
```diff
@@ -1608,14 +1606,14 @@ bool ConditionMgr::isConditionTypeValid(Condition* cond)
...
                          cond->SourceEntry, uint32(cond->ConditionType));
         return false;
     }
     switch (cond->ConditionType) {
-        case CONDITION_TERRAIN_SWAP:
-        case CONDITION_DIFFICULTY_ID:
-            sLog->outErrorDb("SourceEntry %u in `condition` table has a ConditionType that is not supported on 3.3.5a (%u), ignoring.",
+        case CONDITION_UNUSED_41:
+        case CONDITION_UNUSED_49:
+            sLog->outErrorDb("SourceEntry %u in `conditions` table has a ConditionType that is a unused (%u), skipped.",
```
Comment (BarbzYHOOL) [inline src/server/game/Conditions/ConditionMgr.cpp:1616]: same for each "unused" condition

**Comment by BarbzYHOOL** [comment]
hmm or maybe i didn't get exactly, maybe it's not about "not implemented", sorry if so xD but then might need to reword it to make it crystal clear

File: src/server/game/Conditions/ConditionMgr.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/game/Conditions/ConditionMgr.cpp:1273]: Well not in use is something me and Shin made up before.
But not implemented also feels wrong, I'd settle with "is a placeholder" if that sounds ok?

File: src/server/game/Conditions/ConditionMgr.cpp (hunk 2 above)
Comment (Kitzunu) [reply src/server/game/Conditions/ConditionMgr.cpp:1616]: ?

File: src/server/game/Conditions/ConditionMgr.h [hunk 3]
```diff
@@ -132,9 +132,9 @@ enum ConditionSourceType
     CONDITION_SOURCE_TYPE_SMART_EVENT                    = 22,
     CONDITION_SOURCE_TYPE_NPC_VENDOR                     = 23,
     CONDITION_SOURCE_TYPE_SPELL_PROC                     = 24,
-    CONDITION_SOURCE_TYPE_TERRAIN_SWAP                   = 25, // don't use on 3.3.5a
-    CONDITION_SOURCE_TYPE_PHASE                          = 26, // don't use on 3.3.5a
-    CONDITION_SOURCE_TYPE_GRAVEYARD                      = 27, // don't use on 3.3.5a
+    CONDITION_SOURCE_TYPE_UNUSED_25                      = 25, // placeholder
+    CONDITION_SOURCE_TYPE_UNUSED_26                      = 26, // placeholder
+    CONDITION_SOURCE_TYPE_UNUSED_27                      = 27, // placeholder
```
Comment (FrancescoBorzi) [inline src/server/game/Conditions/ConditionMgr.h:137]: why? I think it might be useful to know that these conditions do exist but they are not used in 3.3.5a

Comment (Kitzunu) [reply src/server/game/Conditions/ConditionMgr.h:137]: But why would you keep conditions that do not work for 3.3.5?

Comment (FrancescoBorzi) [reply src/server/game/Conditions/ConditionMgr.h:137]: I'm not saying to keep the condition, just the information that's telling you why that condition is not there

Comment (Kitzunu) [reply src/server/game/Conditions/ConditionMgr.h:137]: So an additional comment?

Comment (FrancescoBorzi) [reply src/server/game/Conditions/ConditionMgr.h:137]: yes

Comment (FrancescoBorzi) [reply src/server/game/Conditions/ConditionMgr.h:137]: but also preserving the original name, why not

Comment (Kitzunu) [reply src/server/game/Conditions/ConditionMgr.h:137]: I'd propose this in that case, so ppl know that they still can use this if they create a new condition
```cpp
CONDITION_SOURCE_TYPE_UNUSED_27                      = 27, // placeholder (CONDITION_SOURCE_TYPE_GRAVEYARD not used in 3.3.5)

Comment (BarbzYHOOL) [reply src/server/game/Conditions/ConditionMgr.h:137]: > why? I think it might be useful to know that these conditions do exist but they are not used in 3.3.5a

?? they are not used in 3.3.5a? because you think conditions = blizzard system?

Comment (FrancescoBorzi) [reply src/server/game/Conditions/ConditionMgr.h:137]: no

Comment (FrancescoBorzi) [reply src/server/game/Conditions/ConditionMgr.h:137]: > I'd propose this in that case, so ppl know that they still can use this if they create a new condition
> 
> ```c++
> CONDITION_SOURCE_TYPE_UNUSED_27                      = 27, // placeholder (CONDITION_SOURCE_TYPE_GRAVEYARD not used in 3.3.5)
> ```

this :)

Comment (Kitzunu) [reply src/server/game/Conditions/ConditionMgr.h:137]: Sure I can add that
